client/reader/sidebar/index.jsx

```diff
@@ -132,42 +137,30 @@ const ReaderSidebar = React.createClass( {
 		}
 	},
 
-	renderAppPromo() {
-		// if promo not configured return
-		if ( ! config.isEnabled( 'desktop-promo' ) ) {
-			return;
-		}
-
-		// if user settings not loaded, return so we dont show
-		// before we can check if user is already a desktop user
-		if ( userSettings.getSetting( 'is_desktop_app_user' ) === null ) {
-			return;
-		}
-
-		// if already using desktop app, dont show promo
-		if ( userSettings.getSetting( 'is_desktop_app_user' ) ) {
-			return;
-		}
-
-		// made it through the gauntlet, show the promo!
-		return (
-			<AppPromo location="reader" />
-		);
+	shouldRenderAppPromo() {
+		return every( [
+			! this.props.isDesktopPromoDisabled,
+			this.props.isUserLocaleEnglish,
+			! this.props.isViewportMobile,
+			! this.props.isChromeOS,
+			this.props.isDesktopPromoConfiguredToRun,
+			! this.props.isUserDesktopAppUser
+		] );
 	},
 
 	render() {
 		return (
```

Comment:
We also like to use some destructuring to remove the long fully-qualified names…

```js
const {
	translate
} = this.props;

… translate( 'Message!' ) …
```